src/cdomains/duplicateVars.ml

```diff
@@ -6,19 +6,20 @@ open CilType
 open GoblintCil
 open Batteries
 open GoblintCil
+module M = Messages
 
 (** Variable Type used by the C-2PO  Analysis.
     It contains normal variables with a varinfo as well as auxiliary variables for
     assignment and return and duplicated variables for remembering the value of variables at the beginning of a function. *)
 module VarType = struct
   (* the hash/compare values should not depend on the type.
      But they have to be defined even though they are not used, for some reason.*)
-  let equal_typ _ _ = true
-  let hash_typ _ = 0
-  let compare_typ _ _ = 0
+  let equal_typ a b = Stdlib.compare a b = 0
+  let hash_typ x = Hashtbl.hash x
+  let compare_typ a b = Stdlib.compare a b
 
-  type t = AssignAux of (typ[@compare.ignore][@eq.ignore][@hash.ignore])
-         | ReturnAux of (typ[@compare.ignore][@eq.ignore][@hash.ignore])
+  type t = AssignAux of typ
+         | ReturnAux of typ
```

Comment:
Is this comment outdated? Because they are used now.

And as such, `CilType.Typ.t` should be used instead such that correct implementations are used. CIL types are cyclic, so `Stdlib.compare` may not terminate.